src/sage/combinat/q_analogues.py

```diff
@@ -328,6 +330,11 @@ def q_binomial(n, k, q=None, algorithm='auto'):
         from sage.rings.polynomial.polynomial_element import Polynomial
         is_polynomial = isinstance(q, Polynomial)
 
+    if n >= 0:
+        k = min(n - k, k)  # Pick the smallest k
+    else:
+        return (-1)**k * q**(k * n - (k * k - k) // 2) * q_binomial(-n + k - 1, k, q=q)
```

Comment:
```suggestion
    if n < 0:
        return (-1)**k * q**(k * n - (k * k - k) // 2) * q_binomial(-n + k - 1, k, q=q)

    k = min(n - k, k)  # Pick the smallest k        
```
More of a stylistic thing, but I find this easier to follow the logic.